### PR TITLE
info.dat: Fully read file

### PR DESCRIPTION
**Pull request by bwrsandman**
Add all Infos structures
Move job of reading Infos to own class.
Fix MagicInfo size by adding 4 bytes.
Re-enable magic type debugging.

This is needed for work being done on villager states in which assigning villagers to abodes needs values from `GAbodeInfo::maxCapacity`.

Relates to #171

### CODE REVIEW

**Comment by raffclar** [comment]
Ready for review?

**Comment by bwrsandman** [comment]
I think @handsomematt wanted to give it a try first. Wait for his approval

**Comment by raffclar** [comment]
> 
> 
> I think @handsomematt wanted to give it a try first. Wait for his approval

Ok

**Review by handsomematt — APPROVED** [review]
Good and important to get in, however not sure if we should mimic exactly how they did the serialization of data in global singleton classes or not, wonder if there are specific reasons they decided to go that route? Won't stop the merge, but worth considering keeping this flexible.

Also I feel like PackFiles are essentially LHSegmentFiles, poorly named?

**Comment by bwrsandman** [comment]
I think it's `LHReleaseFile`

**Pull request merged**
